Code/GraphMol/Wrap/rough_test.py

```diff
@@ -3315,84 +3316,135 @@ def testSDProps(self):
+              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   213',
+               'NSC': 213, 
                'NCI_AIDS_Antiviral_Screen_IC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
-               '_Name': '213', 'CAS_RN': '119-80-2', '_MolFileComments': '119-80-2', 
+               '_Name': 213, 'CAS_RN': '119-80-2', '_MolFileComments': '119-80-2', 
                'NCI_AIDS_Antiviral_Screen_EC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
                'NCI_AIDS_Antiviral_Screen_Conclusion': 'CI'},
-              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   220', 'NSC': '220', 
+              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   220',
+               'NSC': 220, 
                'NCI_AIDS_Antiviral_Screen_IC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
-               '_Name': '220', 'CAS_RN': '5325-83-7', '_MolFileComments': '5325-83-7', 
+               '_Name': 220, 'CAS_RN': '5325-83-7', '_MolFileComments': '5325-83-7', 
                'NCI_AIDS_Antiviral_Screen_EC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
                'NCI_AIDS_Antiviral_Screen_Conclusion': 'CI'},
-              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   229', 'NSC': '229', 
+              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   229',
+               'NSC': 229, 
                'NCI_AIDS_Antiviral_Screen_IC50': '2.00E-04\tM\t>\t2.00E-04\t2', 
-               '_Name': '229', 'CAS_RN': '5325-88-2', '_MolFileComments': '5325-88-2', 
+               '_Name': 229, 'CAS_RN': '5325-88-2', '_MolFileComments': '5325-88-2', 
                'NCI_AIDS_Antiviral_Screen_EC50': '2.00E-04\tM\t>\t2.00E-04\t2', 
                'NCI_AIDS_Antiviral_Screen_Conclusion': 'CI'},
-              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   256', 'NSC': '256', 
+              {'_MolFileInfo': 'BBtclserve11129916382D 0   0.00000     0.00000   256',
+               'NSC': 256, 
                'NCI_AIDS_Antiviral_Screen_IC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
-               '_Name': '256', 'CAS_RN': '5326-06-7', '_MolFileComments': '5326-06-7', 
+               '_Name': 256, 'CAS_RN': '5326-06-7', '_MolFileComments': '5326-06-7', 
                'NCI_AIDS_Antiviral_Screen_EC50': '2.00E-04\tM\t>\t2.00E-04\t4', 
                'NCI_AIDS_Antiviral_Screen_Conclusion': 'CI'},]
     sdSup = Chem.SDMolSupplier(fileN)
     for i,mol in enumerate(sdSup):
       self.assertEquals(mol.GetPropsAsDict(includePrivate=True), sddata[i])
+
+  def testGetSetProps(self):
+    m = Chem.MolFromSmiles("CC")
+    errors = {"int": "key `foo` exists but does not result in a 32 bit integer value",
+              "double": "key `foo` exists but does not result in a 64 bit floating point value",
+              "bool": "key `foo` exists but does not result in a True or False value"}
+
+    for ob in [m, list(m.GetAtoms())[0], list(m.GetBonds())[0]]:
+      ob.SetDoubleProp("foo", 2.0)
+      try:
```

Comment:
you should be using `assertRaises()` here as a context manager so that you can check the exception message:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises